src/prefect/orion/orchestration/core_policy.py

```diff
@@ -529,16 +640,34 @@ async def before_transition(
         proposed_state: Optional[states.State],
         context: FlowOrchestrationContext,
     ) -> None:
+        self.original_flow_policy = context.run.empirical_policy.dict()
```

Comment:
Not blocking for this, but is this the recommend practice for storing state during a rule?

Reply:
I don't know about recommended, but the rule needs to remember the state specifically for cleanup so it's how all of the rules that implement a stateful cleanup do it